streams/notification/src/main/java/com/hortonworks/streamline/streams/notification/store/hbase/mappers/StreamlineEventMapper.java

```diff
@@ -45,13 +45,14 @@
 
     @Override
     public StreamlineEvent entity(Result result) {
-        String id = Bytes.toString(result.getRow());
+        // FIXME: should we preserve event ID here? If then we should enable putting id to event which opens possibility of breaking from other parts.
```

Comment:
@arunmahadevan 
I blocked all the path to assign ID to event, but looks like there's a case which needs to assign ID. This method read rowkey from HBase row and assign it to event's ID. 
Do we want to have some exceptions? Maybe even we want to allow, I still want to make it hard (possible but not easy) to assign ID.

Reply:
@HeartSaVioR , it was done when notification service was initially implemented to query the events associated with a notification. see [here](https://github.com/hortonworks/streamline/blob/master/docs/Notifications.md#get-events-by-id). Right now this is not exposed or used. I'm ok to have default StreamlineEvent implementation not expose a way to pass id. If required we can have some kind of a wrapper event that takes an ID+Event, returns ID and delegates all other methods to underlying event.